### ISSUE TITLE
You are using a plain-text compose editor. You must change to a rich editor to use Markdown Here.

### ISSUE THREAD

**Issue by quchunguang** [adam-p/markdown-here#314]
when i edit a new mail. right-click -> Markdown Toggle. Shows,

```
You are using a plain-text compose editor. You must change to a rich editor to use Markdown Here.
```

**Comment by r2evans** [comment]
I see this message in TB when the account is configured against HTML compositing. For me (TB 47.0, win10), it's under "*Account Settings > Composition & Addressing > Compose message in HTML format*". If this is deselected, you'll get the error you're seeing. Check this box and you shouldn't get the error.

**Rationale**: by default (i.e., people who rely on TB for text formatting in the window, not MDH), when you are entering formatting (bullets, font size, color), behind the scenes it's writing HTML and *rendering* it in a canvas that to you looks like a "*New > Message*" window. If you deselect "compose message in HTML format", then it is being efficient and providing a very simple fixed-width text editor for a new-email window.

There used to be problems with some email viewers rendering too much of the HTML and content, opening security holes to certain types of attacks. From this, some strategies included disallowing HTML in emails and not allowing users to write emails in HTML. (An alternative was rich-text, ala RTF, but that didn't really catch on.)

Essentially, MDH depends on having a page that can provide *rendered* viewing of content, and no-HTML-format in TB does not provide this, so MDH throws the warning.

(Note: I'm not the dev, just an avid user.)

**Comment by Frustrated-Programmer** [comment]
_Account Settings_? as in in github or in the browser? i see no _Compostion & Addressing_ in Github

**Comment by r2evans** [comment]
No, "*Account Settings*" in TB (Thunderbird) version 47.0. The OP started with "edit a new mail", so I was assuming an email client of some form. It's certainly feasible that the question was not about TB, but lacking anything more I included that qualification in my comment.

I don't see what in the original question or my comment would suggest github.

**Comment by Frustrated-Programmer** [comment]
it doesnt :stuck_out_tongue_closed_eyes:

**Comment by Madis0** [comment]
Not using TB or any mail client for that matter, but I was wondering if Markdown Here could work in any text fields by simply _rendering_ them as Markdown? That way it could even work in old Reddit, just as a preview function.

**Comment by r2evans** [comment]
@Madis0, I'm not certain I understand. The point (I believe) is to render *from* markdown *to* another format that is more readily used by the application or read by the recipient(s). This is most commonly HTML, for both email clients and some common rich-text textarea/input-fields.

If you want simple rendering of markdown into something else, see [pandoc](http://pandoc.org/) or it's eventual successor, [commonmark](https://github.com/CommonMark/CommonMark) (still pre-release perhaps?).

**Comment by Madis0** [comment]
@r2evans What I meant is Markdown Here could also support plain text fields which are typed like markdown by previewing their look to user. Unlike Markdown Here's current behaviour, this wouldn't change the typed content, only render it differently.

**Comment by r2evans** [comment]
Plain text fields do not accept any formatting, so you would be significantly altering MDH's behavior. The default (only?) thing it does is convert markdown into HTML within a rich-text-capable field. What you're suggesting is rendering the markdown into a text format, which might preserve things like indents, bullets, and numbered-lists, but would lose things like bold, italics, and hyperlinks.

If that's what you're suggesting, then ... good luck. MDH has received no commits [since Nov 2017](https://github.com/adam-p/markdown-here/commit/6bdab321d3b1415ff99d7dc9bba4f2ebf714bc7d), and with all of the changes to FF and TB recently, I don't know if @adam-p is intending to update or maintain this. (Unfortunately, I'm not able to assist.)

Currently, the only other way I know of other than MDH is to use pandoc/commonmark external to the browser window. As this is a little cumbersome, it is a weak method at best.
